### PR TITLE
Collect kernel version

### LINKED ISSUE #58387: Add kernel version and glibc version to collect_env
## 🚀 Feature

TensorPipe uses some advanced features of the Linux kernel. It does so on a "best effort" basis (i.e., use them if available, fall back on something else if not). This logic is unfortunately hard to test because it would require specific kernel versions on our CI machines. So it occasionally happens that we find out about issues when users report them. To help investigate these reports we've found it useful to know the kernel version and the glibc version used by the user (both when building and when running, if they're different). It would be great if these could be added to the report produced by collect_env.

### PR DESCRIPTION
**Pull request by malfet**
Collect env should collect kernel and glibc version

Fixes #58387

### CODE REVIEW

File: torch/utils/collect_env.py [hunk 1]
```diff
@@ -244,6 +253,17 @@ def get_os(run_lambda):
     # Unknown platform
     return platform
 
+def get_kernel_version():
+    import platform
+    if get_platform() == 'linux':
+        return '{} {}'.format(platform.release(), platform.version())
```
Comment (walterddr) [inline torch/utils/collect_env.py:259]: would 
```
platform.platform()
```
gives more info?

File: torch/utils/collect_env.py [hunk 2]
```diff
@@ -244,6 +253,17 @@ def get_os(run_lambda):
...
+    else:
+        return platform.release()
+
+def get_libc_version(run_lambda):
```
Comment (walterddr) [inline torch/utils/collect_env.py:263]: there's also `platform.libc_ver()` if that works better?

**Review by walterddr — APPROVED** [review]

File: torch/utils/collect_env.py (hunk 1 above)
Comment (malfet) [reply torch/utils/collect_env.py:259]: Good point, although it's not just a kernel version and can be a bit confusing, for example I have glibc-2.31 installed on my system, but `platform.platform()` reports glibc-2.2.5:
```
$ python -c "import platform;print(platform.platform());";/lib64/libc.so.6|head -n1
Linux-5.11.12-100.fc32.x86_64-x86_64-with-glibc2.2.5
GNU C Library (GNU libc) stable release version 2.31.
```

Comment (walterddr) [reply torch/utils/collect_env.py:259]: yeah I was about to ask this question --> which libc do we actually care about?
for example in the conda environment there's a glibc that's different from non-conda env
```
Linux-5.2.9-229_fbk15_hardened_4185_g357f49b36602-x86_64-with-glibc2.28 -- from conda python3
Linux-5.2.9-229_fbk15_hardened_4185_g357f49b36602-x86_64-with-glibc2.10 -- from native python3
```

File: torch/utils/collect_env.py (hunk 2 above)
Comment (walterddr) [reply torch/utils/collect_env.py:263]: echo on the comment https://github.com/pytorch/pytorch/pull/58485#discussion_r634573136.

**Review by walterddr — APPROVED** [review]

**Comment by lw** [comment]
Thanks!

**Pull request closed without merging**